src/Connection/UnprocessedRequestException.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Amp\Http\Client\Connection;
+
+use Amp\Http\Client\HttpException;
+use Amp\Http\Client\Request;
+
+final class UnprocessedRequestException extends HttpException
+{
+    /** @var Request */
+    private $request;
+
+    public function __construct(Request $request, HttpException $previous)
+    {
+        parent::__construct("The request was not processed and can be safely retried", 0, $previous);
+        $this->request = $request;
+    }
+
+    public function getRequest(): Request
+    {
+        return clone $this->request;
+    }
```

Comment:
Maybe we should move this method and `isRetryable()` into `HttpException`?

Reply:
Not everything extending `HttpException` necessarily has a `Request` object to associate with it. Maybe another exception extending `HttpException` that I can use as a base between the two?